### DISCUSSION TITLE
Welcome to proctoring Discussions!

### DISCUSSION THREAD

**Discussion by sanket2000** [sanket2000/proctoring#3] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by akashvardhan02** [comment]


**Reply by sanket2000** [reply]
Follow these steps.

Install python on your system.
Then execute these commands,
1. git clone https://github.com/sanket2000/proctoring.git
2. cd proctoring
3. pip install -r requirements.txt
4. python run.py

Please share some screenshots or error text in any step

**Reply by sanket2000** [reply]
Change your current working directory to "proctoring"

**Reply by sanket2000** [reply]
1. cd src
2. python run.py

**Reply by sanket2000** [reply]
To terminate the program you have to kill the terminal.

Always read the error carefully, the error indicates that the camera index is out of range. See how you can make it use your camera. Try simple opencv camera read code and figure out what is the camera index, then replace the same index at head_pose.py file line 24.

It's too late today to continue the discussion, I suggest to do more trial and error and solve such basic problems.
I'll keep an eye on the thread, and respond as I get some spare time
All the best

**Comment by akashvardhan02** [comment]


**Reply by sanket2000** [reply]
I assume your python has some faults, or you may have manually deleted some python core files, 
reinstall python once and try again.
Use python 3.10 or higher if possible.
Avoid python 3.12 for now

**Reply by sanket2000** [reply]
Steps are
Go to control panel
Uninstall python
Got to https://www.python.org/downloads/release/python-3100/
And download Windows installer (64-bit)
Install python with default settings, and check "Add to path" option in installation.

**Reply by sanket2000** [reply]
Yes git clone, pip install etc do that

**Reply by sanket2000** [reply]
In requirements.txt make all versions as == instead of >=

**Reply by sanket2000** [reply]
﻿mediapipe==0.8.9 
opencv-python==4.5.5 
sounddevice==0.4.4 
virtualenv==20.13

**Reply by sanket2000** [reply]
Your python has problem that is for sure delete python37 folder or uninstall that python 3.7 then my previous steps will help.

**Comment by sanket2000** [comment]
Why did you delete the discussion, it could have been helpful for others.

**Reply by akashvardhan02** [reply]
beacause my lecture will verify the work  bro sorry for that

**Comment by PrathamHandique** [comment]
I am using mac os and getting this error,says that plt.show() and plt.gca()  should not be outside the main thread .

(env) (base) prathamhandique@Prathams-MacBook-Air proctoring % python src/run.py
/Users/prathamhandique/Desktop/proctoring/src/detection.py:87: UserWarning: Starting a Matplotlib GUI outside of the main thread will likely fail.
  plt.show()
/Users/prathamhandique/Desktop/proctoring/src/detection.py:88: UserWarning: Starting a Matplotlib GUI outside of the main thread will likely fail.
  axes = plt.gca()
  
  INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'NSWindow should only be instantiated on the main thread!'
*** First throw call stack:

**Reply by sanket2000** [reply]
Since the project is little old, and the dependent library must have been upgraded, I suggest you to install the python dependencies to the exact version mentioned in requirements.txt 

I suspect it might be because of latest version of matplot lib

Just create new venv and install requirements.txt with == for version match 

Like

abc==2.3.1
Instead of 
abc>=2.3.1

**Comment by AryanDude** [comment]
Bro how can i find the project accuracy. My teacher wants me to find it and compare with other models

**Reply by sanket2000** [reply]
Thanks for using this discussion page, it's helpful for everyone using this project.

one way of finding accuracy is to find accuracy of the base model being used. In this project we used google mediapipe model to detect face in real time. So you can check that for accuracy of face detection.

And if you are thinking of accuracy of project as whole, it is difficult to define due to lack of real world data. It will be very time consuming to collect and label such data. You can put these arguments with you teacher to try to make them understand.

**Reply by AryanDude** [reply]
Ok thank you brother.  Then i write accuracy of audio and video feed separately in my report.

**Reply by AryanDude** [reply]
Bro I wanted to ask something else. How did you set the weights for x cheat, audio cheat etc

**Reply by sanket2000** [reply]
Manually we did trial and error and then set the weights. First we thought of ML approch but getting high quality labeled data was difficulty.
